### PR TITLE
[Merged by Bors] - Feat: prove `IsTrans α r → Trans r r r` and `Trans r r r → IsTrans α r`

### PR DESCRIPTION
**Pull request by urkud**
Now `Trans.trans` conflicts with `_root_.trans`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Init/Algebra/Classes.lean [hunk 1]
```diff
@@ -177,6 +177,12 @@ class IsAntisymm (α : Type u) (r : α → α → Prop) : Prop where
 class IsTrans (α : Type u) (r : α → α → Prop) : Prop where
   trans : ∀ a b c, r a b → r b c → r a c
 
+instance {α : Type u} {r : α → α → Prop} [IsTrans α r] : Trans r r r :=
+  ⟨IsTrans.trans _ _ _⟩
+
+instance {α : Type u} {r : α → α → Prop} [Trans r r r] : IsTrans α r :=
+  ⟨fun _ _ _ => Trans.trans⟩
+
```
Comment (kbuzzard) [inline Mathlib/Init/Algebra/Classes.lean:185]: Expecting the answer "yes": so these looping instances are OK?

**Review by kbuzzard — COMMENTED** [review]
The PR looks fine but why are we making this change? Is there a Zulip discussion or something?

File: Mathlib/Init/Algebra/Classes.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Init/Algebra/Classes.lean:185]: @digama0 said yes.

**Comment by urkud** [comment]
> The PR looks fine but why are we making this change? Is there a Zulip discussion or something?

I'm making this PR because I want to write `(h : l ⊆ l').trans _`.

**Comment by fpvandoorn** [comment]
bors merge

**Comment by fpvandoorn** [comment]
please merge master and fix conflicts after bors is done with the current run

bors r-
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**